### ISSUE TITLE
missing field `network` at line 1 column 18453.

### ISSUE THREAD

**Issue by perseus-algol** [input-output-hk/mithril#1651]
## Context & versions

Latest version at [820d9377e67c4e05ad3614f7b66b3e15249f5656](https://github.com/input-output-hk/mithril/commit/820d9377e67c4e05ad3614f7b66b3e15249f5656) commit.

## Steps to reproduce

It is possible to reproduce the problem either using [explorer](https://mithril.network/explorer/?aggregator=https%3A%2F%2Faggregator.release-mainnet.api.mithril.network%2Faggregator), or mithril-client-cli.

The problem appears only on: `release-mainnet`, `release-preprod`, `pre-release-preview` cardano networks. At `testing-preview` and `testing-sanchonet` everything works fine. 

### To reproduce problem using `explorer`

- open https://mithril.network/explorer/, 
- choose `release-mainnet` (or any first 3 networks), 
- open browser DevTools (Ctrl+Shift+I), open console tab
- click "Show" on any DB snapshot
- Click Verify.

Actual result: 
- Popup window "Verify certificate" has empty content.
- You will get an error in browser console: `VerifyCertificateModal init error: missing field `network` at line 1 column 18453`.

Expected result: Verification process started. In Popup window "Verify certificate" you should see certificates chain.

### To reproduce using `mithril-client-cli`

Prepare `mithril-client-cli/.env` file with mainnet configuration:

```bash
# Cardano network
export NETWORK=mainnet

# Aggregator API endpoint URL
export AGGREGATOR_ENDPOINT=https://aggregator.release-mainnet.api.mithril.network/aggregator

# Genesis verification key
export GENESIS_VERIFICATION_KEY=$(wget -q -O - https://raw.githubusercontent.com/input-output-hk/mithril/main/mithril-infra/configuration/release-mainnet/genesis.vkey)

# Digest of the latest produced snapshot for convenience of the demo
# You can also modify this variable and set it to the value of the digest of a snapshot that you can retrieve at step 2
export SNAPSHOT_DIGEST=latest
```

Then in terminal:

```bash
cd mithril-client-cli
source .env
cargo run -- cardano-db download 3aeb777c894d330e48e7136d3101b2576f6f956645957a9bd0bbb0ca00f62a48  # replace hash with existing one
```

Actial result: 

```
Apr 23 16:33:46.455 CRIT Could not create certificate from API message: missing field `network` at line 1 column 18453.
Error: Can not verify the certificate chain from certificate_hash: '8383aebb521aae29b2100d0c776ea5567a66943534de4f310a4c884af33f0dce'

Caused by:
    missing field `network` at line 1 column 18453
```

@jpraynaud, you can assign it on me.

**Comment by perseus-algol** [comment]
So I see the cause of the problem. Aggregator's endpoint `GET /certificate/{certificate_hash}` returns `network` field in cert metadata at sanchonet, and mainnet aggregator doesn't. 

In [this commit](https://github.com/input-output-hk/mithril/commit/ed0ebbd7894b310579653667ffd2d30f99ca6e46#diff-08a79ee7dfe8196367af6fee405e19491ef525a0c261a6223bf5fc8a84febea6R10) `network` field was added to `CertificateMetadataMessagePart` on 28 Mar 2024. So I suspect that there is an older version of aggregator deployed with mainnet. And latest version deployed with sanchonet.

**Comment by jpraynaud** [comment]
@perseus-algol thanks for creating the issue and investigating on it!

I think that this is related to a [breaking change](https://github.com/input-output-hk/mithril/blob/main/CHANGELOG.md?plain=1#L12) introduced which makes the `unstable` version of the client (the one which is deployed with the explorer) compatible only with the `testing-preview` and `tetsing-sanchonet` aggregators (not any more with the `pre-release-preview`, `release-preprod` or `release-mainnet` aggregators which are not running the `latest` version).

This is a temporary situation and the certificate verification will resume on all the networks when the next distribution is released :+1:

**Comment by jpraynaud** [comment]
The problem has been fixed with the release of the new [`2418.1`](https://github.com/input-output-hk/mithril/releases/tag/2418.1) distribution and the explorer is now able to verify the certificates from all the Mithril networks 👍
